### PR TITLE
Feature facade

### PR DESCRIPTION
**Pull request by GustavoZHA**
Facade implement to communicate controller and service

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java [hunk 1]
```diff
@@ -20,13 +18,13 @@
 import org.springframework.web.bind.annotation.RestController;
 import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
-
 import java.io.IOException;
 import java.security.NoSuchAlgorithmException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: This blank line should be deleted

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 2]
```diff
@@ -10,17 +10,12 @@
...
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.fundacion.jala.converter.service.ExtractMetadata;
+import org.fundacion.jala.converter.models.facade.ExtractFacade;
 import org.fundacion.jala.converter.service.FileStorageService;
-import org.fundacion.jala.converter.service.ObjectMetadata;
-import org.fundacion.jala.converter.service.metadata.TypeFileExport;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 3]
```diff
@@ -10,17 +10,12 @@
...
-import org.springframework.web.multipart.MultipartFile;
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: This blank line should be deleted

File: src/main/java/org/fundacion/jala/converter/controller/ExtractTextController.java [hunk 4]
```diff
@@ -39,11 +38,8 @@ public String uploadFile(final @RequestParam("file")MultipartFile file,
                              final @RequestParam("language") String language,
                              final @RequestParam("nameOutput") String nameOutput) throws IllegalStateException, IOException {
         LOGGER.info("start");
-        String filename = file.getOriginalFilename();
-        String storagePath = fileStorageService.uploadFile(file);
-        ExtractText extractText = new ExtractText(new ExtractTextParameter(storagePath, language, nameOutput));
-        extractText.extractText();
-        String outputPath = FileStorageService.getOutputPath(filename);
+
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractTextController.java]: This blank line should be deleted

File: src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java [hunk 5]
```diff
@@ -31,10 +31,12 @@
     public String compileJava(final @RequestParam("code") String code) throws IllegalStateException, IOException {
         LOGGER.info("start");
         if (!code.isBlank() || !code.equals(null)){
-            JavaCompiler javaCompiler = new JavaCompiler();
+            CompilerFacade java = new CompilerFacade();
+//            JavaCompiler javaCompiler = new JavaCompiler();
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java]: This commented line should be removed

File: src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java [hunk 6]
```diff
@@ -31,10 +31,12 @@
...
         LOGGER.info("start");
         if (!code.isBlank() || !code.equals(null)){
-            JavaCompiler javaCompiler = new JavaCompiler();
+            CompilerFacade java = new CompilerFacade();
+//            JavaCompiler javaCompiler = new JavaCompiler();
             String filePath = Transform.toFile(code, "Main", "java");
             LOGGER.info("finish");
-            return javaCompiler.javaCompiler(JavaVersion.JAVA_11, filePath);
+            return java.facadeJavaCompile(JavaVersion.JAVA_11,filePath);
+//            return javaCompiler.javaCompiler(JavaVersion.JAVA_11, filePath);
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java]: This commented line should be removed

File: src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java [hunk 7]
```diff
@@ -10,12 +10,10 @@
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-import org.fundacion.jala.converter.service.FileStorageService;
+import org.fundacion.jala.converter.models.facade.CompilerFacade;
 import org.fundacion.jala.converter.service.pythoncompiler.Python;
-import org.fundacion.jala.converter.service.pythoncompiler.*;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.web.bind.annotation.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java [hunk 8]
```diff
@@ -30,10 +28,12 @@
     public String compilePython(final @RequestParam("code") String code) throws IllegalStateException, IOException {
         LOGGER.info("start");
         if (!code.isBlank() || !code.equals(null)){
-            PythonCompiler pythonCompiler = new PythonCompiler();
+//            PythonCompiler pythonCompiler = new PythonCompiler();
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java]: This commented line should be removed

File: src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java [hunk 9]
```diff
@@ -30,10 +28,12 @@
...
         LOGGER.info("start");
         if (!code.isBlank() || !code.equals(null)){
-            PythonCompiler pythonCompiler = new PythonCompiler();
+//            PythonCompiler pythonCompiler = new PythonCompiler();
+            CompilerFacade python = new CompilerFacade();
             String filePath = Transform.toFile(code, "filetocompile", "py");
             LOGGER.info("finish");
-            return pythonCompiler.compiler(Python.V3, filePath);
+          return python.facadePythonCompile(Python.V3, filePath);
+//            return pythonCompiler.compiler(Python.V3, filePath);
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java]: This commented line should be removed

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java [hunk 10]
```diff
@@ -17,6 +17,8 @@
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java]: This blank line should be removed

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java [hunk 11]
```diff
@@ -1,6 +1,6 @@
 /**
  * Copyright (c) 2021 Fundacion Jala.
- * <p>
+ *
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:3]: You should only import what is necessary, not the entire library "*"
Line 15

File: src/main/java/org/fundacion/jala/converter/models/facade/ExtractFacade.java [hunk 12]
```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
+ * <p>
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/models/facade/ExtractFacade.java]: This "< p >" should be deleted

**Review by JRojas999 — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java (hunk 1 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/AudioConverterController.java]: solve

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 2 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: solve

File: src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java (hunk 5 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java]: solved

File: src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java (hunk 6 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/JavaCompilerController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java [hunk 13]
```diff
@@ -35,77 +30,14 @@
...
-        ExtractMetadata extractMetadata = new ExtractMetadata(new File(fileToExtract), new File(fileToExport));
-        extractMetadata.extractMetadata();
-        File file = new File(fileToExtract);
-        String filename = file.getName();
-        String storagePath = filename;
-        String outputPath = FileStorageService.getOutputPath(filename);
+        String filename =ExtractFacade.getMetadataExtract(fileToExtract,isMoreInfo,nameExport,format);
         final String baseUrl = ServletUriComponentsBuilder.fromCurrentContextPath().build().toUriString();
-        String downloadLink = baseUrl + "/api/download/" + filename;
+        String downloadLink = baseUrl + "/api/download/" + filename+"."+format;
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: Blanks are needed between the "+" signs

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java [hunk 14]
```diff
@@ -17,6 +17,8 @@
 import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
 import java.io.IOException;
+
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java]: This blank line should be removed

File: src/main/java/org/fundacion/jala/converter/models/facade/CompilerFacade.java [hunk 15]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
+ * <p>
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/models/facade/CompilerFacade.java]: This "< p >" should be deleted

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 3 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java (hunk 13 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/ExtractMetadataController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java (hunk 8 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java]: Ok

File: src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java (hunk 9 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java]: You are rigth

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java (hunk 11 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java:3]: ok

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java (hunk 10 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java]: ok

File: src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java (hunk 14 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/VideoConverterController.java]: ok

File: src/main/java/org/fundacion/jala/converter/models/facade/CompilerFacade.java (hunk 15 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/models/facade/CompilerFacade.java]: oks

File: src/main/java/org/fundacion/jala/converter/models/facade/ExtractFacade.java (hunk 12 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/models/facade/ExtractFacade.java]: ok dani

File: src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java (hunk 7 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/controller/PythonCompilerController.java]: ok

**Pull request closed without merging**